server/controllers/users.js

```diff
@@ -59,20 +62,22 @@ class UserController {
       const errorValue = error.details[0].message.replace(/\"/g, '');
       return res.status(400).json({ status: 400, error: errorValue });
     }
-    passport.authenticate('local', { session: false }, (
-      err,
-      user,
-      info
-    ) => {
-      if (err) {
-        return next(err);
-      }
-
-      if (user) {
-        return res.json({ user });
-      }
-      return res.status(400).json(info);
-    })(req, res, next);
+    try {
+      const { email, password } = req.body;
+      const user = await User.findOne({
+        where: {
+          email,
+        }
+      });
+      if (!user) return res.status(400).json({ status: 400, error: 'Invalid email or password' });
+      // verify password
+      const match = await bcrypt.compare(password, user.password);
+      if (!match) return res.status(400).json({ status: 400, error: 'Invalid email or password' });
+      const token = await Token.create(user);
```

Comment:
You are creating a token with the entire user object?

tokens are not encrypted. they are just a `base64` encoding of the object and a signed checksum of the object.

What you have here exposes our users sensitive details like password (even though this is hashed, it's still a security risk).

You are okay (for now with encoding only the users `id` and `username`)

Reply:
In the token class, I have extracted the id and email from the user. I am encrypting only the id and email. Is that okay?

Reply:
no.

You may have reason to reuse the `Token.create` method.

sending only what you want to encode to it would make it easier to reuse

Reply:
ok. I understand